hs_tracking/models.py

```diff
@@ -18,31 +18,38 @@
 
 
 class SessionManager(models.Manager):
-    def for_request(self, request):
+    def for_request(self, request, user=None):
+        if hasattr(request, 'user'):
+            user = request.user
+
         signed_id = request.session.get('hs_tracking_id')
         if signed_id:
             tracking_id = signing.loads(signed_id)
             cut_off = datetime.now() - timedelta(seconds=SESSION_TIMEOUT)
             session = None
+
             try:
                 session = Session.objects.filter(
                     variable__timestamp__gte=cut_off).filter(id=tracking_id['id']).first()
             except Session.DoesNotExist:
                 pass
+
             if session is not None:
-                if session.visitor.user is None and request.user.is_authenticated():
+                if session.visitor.user is None and user.is_authenticated():
```

Comment:
@jbradberry OK. Got it. From code logic point of view and to make code more robust, perhaps addition of a simple conditional check on line 37 to make it like ```if session is not None and user is not None``` is better?